File: 卡码/代码随想录/Python/数组/04-长度最小的子数组/LeetCode_0209.py
#209. 长度最小的子数组
from typing import List

class Solution:
	def minSubArrayLen(self, target: int, nums: List[int]) -> int:
		i,sum,res=0,0,len(nums)+1
		for j in range(len(nums)):
			sum+=nums[j]
			while sum>=target:
				res=min(res,j-i+1)
				sum-=nums[i]
				i+=1
		return res if res!=len(nums)+1 else 0

def main():
	target=7
	nums=[2,3,1,2,4,3]
	solution=Solution()
	print(solution.minSubArrayLen(target,nums))

if __name__ == '__main__':
	main()

File: 卡码/代码随想录/1/01-数组/C++/04-长度最小的子数组/LeetCode_0209.cpp
//209. 长度最小的子数组
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
  int minSubArrayLen(int target, vector<int>& nums) {
		int i=0,sum=0,res=nums.size()+1;
		for(int j=0;j<nums.size();j++){
			sum+=nums[j];
			while(sum>=target){
				res=min(res,j-i+1);
				sum-=nums[i];
				i++;
			}
		}
		return res==nums.size()+1 ? 0 : res;
  }
};

int main(){
	int target=7;
	vector<int> nums={2,3,1,2,4,3};
	Solution* solution=new Solution();
	printf("%d\n",solution->minSubArrayLen(target,nums));
	return 0;
}